Automattic/sensei
# Only enqueue user-dashboard.js where required.

Fixes #2798 

#### Changes proposed in this Pull Request:

* Only enqueue the `user-dashboard.js` file for the
"Learner profile" page and `[usercourses]` shortcode. This should help UI performance
for the pages that don't require `user-dashboard.js`
and its dependencies.

#### Testing instructions:

* Visit the any page in a Sensei LMS installation that is not a learner profile page or uses the `[usercourses]` shortcode, i.e. "My Courses". Check the browser's developer tools to see that `wp-content/plugins/sensei/assets/js/user-dashboard.min.js` is not loaded. The course categories should appear correctly as UI tabs.
* Visit a Learner Profile page (`/learner/<username>`). The user's courses should appear as tabs (see screenshot) and using the browser's developer tools, you should see that `wp-content/plugins/sensei/assets/js/user-dashboard.min.js` is loaded. 

<img width="497" alt="Screen Shot 2020-02-17 at 3 35 03 PM" src="https://user-images.githubusercontent.com/4930129/74685280-08a4fd80-519c-11ea-9a9a-10777d0adef4.png">

* Create a new page with the shortcode `[usercourses]` in it. The user's courses should appear as tabs below the deprecation notice (see screenshot) and using the browser's developer tools, you should see that `wp-content/plugins/sensei/assets/js/user-dashboard.min.js` is loaded. 

<img width="625" alt="Screen Shot 2020-02-17 at 3 34 12 PM" src="https://user-images.githubusercontent.com/4930129/74685350-4dc92f80-519c-11ea-8f25-67f163d948de.png">



#### Screenshot / Video




#### Proposed changelog entry for your changes:
* Only load `user-dashboard.js` when required.
